src/mpid/ch4/src/ch4_persist.c

```diff
@@ -165,3 +165,345 @@ int MPID_Recv_init(void *buf,
   fn_fail:
     goto fn_exit;
 }
+
+int MPID_Bcast_init(void *buffer, MPI_Aint count, MPI_Datatype datatype,
+                    int root, MPIR_Comm * comm_ptr, MPIR_Info * info_ptr, MPIR_Request ** request)
+{
+    int mpi_errno = MPI_SUCCESS;
+    MPIR_FUNC_ENTER;
+
+    mpi_errno = MPIR_Bcast_init_impl(buffer, count, datatype, root, comm_ptr, info_ptr, request);
```

Comment:
You may just delete the custom code in `src/binding/coll_api.txt` and the binding layer should generate code to call `MPIR_Bcast_init_impl` directly.

Reply:
I'll split that change to a separate PR since the request stuff does not depend on it. That way others can comment on it and we can merge before discussion of this is resolved.

Reply:
On closer look the calls to `MPID_Bcast_init` and friends are generated by `maint/gen_coll.py`. There is at least one instance where the `MPIR_*_init` functions do more than just directly call the device (which directly calls the `MPIR_*_init_impl` layer). That is, for reduce collectives where host buffers are allocated for doing reductions on GPU data, if necessary.

Reply:
I was wrong. All collectives have device layer so vendor can have their own algorithms. Even though most of the device layer just comm MPIR_xxx_impl, they are still needed.